plugins/plugin-languageserver/che-plugin-languageserver-ide/src/main/java/org/eclipse/che/plugin/languageserver/ide/editor/quickassist/LanguageServerQuickAssistProcessor.java

```diff
@@ -75,26 +78,26 @@ public void respond(
               Map<Annotation, org.eclipse.che.ide.api.editor.text.Position> annotations) {
             // iteration with range never returns anything; need to filter ourselves.
             // https://github.com/eclipse/che/issues/4338
-            List<Diagnostic> diagnostics =
-                annotations
-                    .entrySet()
-                    .stream()
-                    .filter(
-                        (e) -> e.getValue().overlapsWith(range.getStartOffset(), range.getLength()))
-                    .map(Entry::getKey)
-                    .map(a -> (DiagnosticAnnotation) a)
-                    .map(DiagnosticAnnotation::getDiagnostic)
-                    .collect(Collectors.toList());
+            List<Diagnostic> diagnostics = new ArrayList<>();
```

Comment:
Probably it would be better to use `java.util.LinkedList` here and further as we don't know the exact size of a list, however I don't think this is so critical.

Reply:
I don't see it as something that can give a big improvement as in reality this list doesn't consist like hundreds of elements - in my case it has one element while there are four error annotations in the same line. We just need to store "more than one" diagnostics here as some error annotations might overlap... Not more, so it's not about the performance, imho.